clarity/src/vm/analysis/type_checker/mod.rs

```diff
@@ -741,12 +972,88 @@ impl<'a, 'b> TypeChecker<'a, 'b> {
         }
     }
 
+    fn inner_type_check_expects(
+        &mut self,
+        expr: &SymbolicExpression,
+        context: &TypingContext,
+        expected_type: &TypeSignature,
+    ) -> TypeResult {
```

Comment:
Just want to clarify -- this can _only_ be called when analyzing Clarity 2 code, right?  Can there be a defensive check here?

Reply:
Correct. It is only called from the branch of `type_check_expects` where `self.clarity_version >= ClarityVersion::Clarity2`. We can add another check here if you'd like.

Reply:
Instead of having a check in both the caller and a check in the callee, how about I rename this function to make it clear that it is for clarity2? `clarity2_inner_type_check_expects`? Or I could just inline the code into `type_check_expects`.

Reply:
I think the best option would be to name this `clarity2_type_check_expects`, then name the previous `type_check_expects` as `clarity1_type_check_expects`, and then `type_check_expects` would be a new function that just switches on the `ClarityVersion` and returns the appropriate `type_check_expects` implementation.

Reply:
Refactored in 2d88f80eb.